src/frontend/src/types/apptypes.ts

```diff
@@ -249,8 +249,19 @@ export type FilterDefinition<MetadataType = any, ValueType = any> = {
 	defaultDescription?: string;
 	/** Name of the component, for filters generated from the blacklab index metadata, `filter-${uiType}` */
 	componentName: string;
+
+	/** If defined, overrides the component name when determining how the widget should behave.
+	 *  (by default, all widgets behave as document-level metadata widgets)
+	 */
+	behaviourName?: string;
+
+	/** Is this a span-based filter, i.e. selecting parts documents instead of entire documents?
+	 *  (i.e. filter on speaker in documents with multiple speakers; done using a "within" BCQL query) */
+	isSpanFilter?: boolean;
```

Comment:
Even filosoferen hoor... 
isSpanFilter geeft alleen maar aan wat voor soort implementatie er zit achter de (behaviorName || componentName) toch? Een filterDefinition met `behaviorName: undefined` en `componentName: filter-text`  (een standaard text input dus als ik het goed begrijp) met `isSpanFilter: true` zou onzin zijn toch (want daar komt nooit iets uit behalve een basic lucene query)? Zou het dan niet beter zijn als `isSpanFilter` een property in de filterValueFunctions zou zijn?

Reply:
Yes, had ook al in m'n achterhoofd dat dit een beetje overdefined was. Fixed.